zerver/decorator.py

```diff
@@ -164,28 +164,30 @@ def validate_api_key(role, api_key):
     return profile
 
 # Use this for webhook views that don't get an email passed in.
-def api_key_only_webhook_view(view_func):
-    @csrf_exempt
-    @has_request_variables
-    @wraps(view_func)
-    def _wrapped_view_func(request, api_key=REQ,
-                           *args, **kwargs):
+def api_key_only_webhook_view(client_name):
+    def _wrapped_view_func(view_func):
+        @csrf_exempt
+        @has_request_variables
+        @wraps(view_func)
+        def _wrapped_func_arguments(request, api_key=REQ,
+                                    *args, **kwargs):
 
-        try:
-            user_profile = UserProfile.objects.get(api_key=api_key)
-        except UserProfile.DoesNotExist:
-            raise JsonableError("Invalid API key")
-        if not user_profile.is_active:
-            raise JsonableError("Account not active")
-        if user_profile.realm.deactivated:
-            raise JsonableError("Realm for account has been deactivated")
-
-        request.user = user_profile
-        request._email = user_profile.email
-        process_client(request, user_profile)
-        if settings.RATE_LIMITING:
-            rate_limit_user(request, user_profile, domain='all')
-        return view_func(request, user_profile, *args, **kwargs)
+            try:
+                user_profile = UserProfile.objects.get(api_key=api_key)
+            except UserProfile.DoesNotExist:
+                raise JsonableError("Invalid API key")
+            if not user_profile.is_active:
+                raise JsonableError("Account not active")
+            if user_profile.realm.deactivated:
+                raise JsonableError("Realm for account has been deactivated")
+
+            request.user = user_profile
+            request._email = user_profile.email
+            process_client(request, user_profile)
+            if settings.RATE_LIMITING:
+                rate_limit_user(request, user_profile, domain='all')
+            return view_func(request, user_profile, get_client(client_name), *args, **kwargs)
```

Comment:
I think the `process_client` logic should be replaced here with something that logs based on client_name -- apparently we were not logging the webhook client properly.